### PR TITLE
Bump version to 1.16.0-dev

### PR DESCRIPTION
**Pull request by rajaths010494**
Since 1.15.1 is released, so bumped up the version to 1.16.0-dev

### CODE REVIEW

File: astronomer/providers/package.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ def get_provider_info() -> Dict[str, Any]:
         "package-name": "astronomer-providers",
         "name": "Astronomer Providers",
         "description": "Apache Airflow Providers containing Deferrable Operators & Sensors from Astronomer",
-        "versions": "1.15.1",
+        "versions": "1.16.0.dev1",
```
Comment (pankajkoti) [inline astronomer/providers/package.py:11]:
```suggestion
        "versions": "1.16.0dev",
```

File: docs/conf.py [hunk 2]
```diff
@@ -27,7 +27,7 @@
 author = "Astronomer Inc."
 
 # The full version, including alpha/beta/rc tags
-release = "1.15.1"
+release = "1.16.0.dev1"
```
Comment (pankajkoti) [inline docs/conf.py:30]:
```suggestion
release = "1.16.0dev"
```

File: setup.cfg [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = astronomer-providers
-version = 1.15.1
+version = 1.16.0.dev1
```
Comment (pankajkoti) [inline setup.cfg:3]:
```suggestion
version = 1.16.0dev
```

**Review by pankajkoti — COMMENTED** [review]
I have not observed `.` between the patch version and suffixes. 
And since we won't have `dev1`, `dev2`, `dev3` out, just `dev` is sufficient. If we want to create RCs, they can be `rc1`, `rc2`, `rc3`, etc. e.g. `1.16.0rc1`

So I believe we can start following this convention unless we have other opinions and examples of projects with semantic versioning following such a pattern.

File: astronomer/providers/package.py (hunk 1 above)
Comment (rajaths010494) [reply astronomer/providers/package.py]: Changed

File: docs/conf.py (hunk 2 above)
Comment (rajaths010494) [reply docs/conf.py]: Changed

File: setup.cfg (hunk 3 above)
Comment (rajaths010494) [reply setup.cfg]: Changed

**Review by pankajkoti — APPROVED** [review]
Please also update the description of the PR as per the latest commit. Thanks for the bump!

**Pull request merged**